.github/workflows/ui-tests-critical.yml

```diff
@@ -82,7 +82,7 @@ jobs:
         with:
           name: ${{env.APP_ARTIFACT_NAME}}
           path: ${{env.APP_PATH}}
-      - uses: futureware-tech/simulator-action@bfa03d93ec9de6dacb0c5553bbf8da8afc6c2ee9 # pin@v3
+      - uses: futureware-tech/simulator-action@dab10d813144ef59b48d401cd95da151222ef8cd # pin@v4
```

Comment:
`m`: I overlooked this when reviewing the PR for adding this code. @krystofwoldrich, I would prefer to replace this GH action with create-simulator.sh. What made you choose this GH action, @krystofwoldrich?

https://github.com/getsentry/sentry-cocoa/blob/f69c05d63f419618670f62a87eca9881c43d6c6f/.github/workflows/ui-tests-critical.yml#L75

Reply:
`create-simulator.sh` creates a new sim device. 

This active clear the device, finds it based the options, starts it and awaits boot.

We could replace it by `start-simulator.sh`, but I don't see why to implement this again.

Reply:
Can we also do this quickly ourselves with bash? Every GH action is a security risk for leaking GH secrets. If not, then I think it's OK to keep it.